compensations/src/main/java/org/jboss/narayana/compensations/impl/ParticipantImpl.java

```diff
@@ -92,9 +94,10 @@ public void close() throws WrongStateException, SystemException {
             ConfirmationHandler handler = instantiate(confirmationHandler);
             handler.confirm();
 
-            CompensationContext.close(currentTX);
             Thread.currentThread().setContextClassLoader(origClassLoader);
         }
+
+        CompensationContextLifetimeManager.getInstance().participantDestroyed(currentTX);
```

Comment:
What exception do you have in mind? CompensationContextLifetimeManager class comes with this PR